### DISCUSSION TITLE
Constraint modeling, Illegal use of "for"

### DISCUSSION THREAD

**Discussion by sb1368** [yalmip/YALMIP#1429] [Optimization modelling] [answered]
Hello,
Thanks for your answering,

How can I model the following constraint in YALMIP?

sub(i,j,c)-s(i,c)  <= 0;     for all i,j,c

I examined the following code but it is not true (obj is not important) and I get the error "Illegal use of reserved keyword "for"":

clc
clear

I=40;J=40;C=15;

s=sdpvar(I,C);
sub=sdpvar(I,J,C);
subp=sdpvar(I,C);

con=[for k=1:J
          subp(:,:)=sub(:,k,:);
          subp-s<=0;
     end
    ]
 obj=sum(s)-sum(sub);
 sett=sdpsettings('solver','sedumi');
 optimize(con,obj,sett);
    
 Cost=value(obj)

How can I add the mentioned constraint to my optimization model?

**Comment by johanlofberg** [comment, marked as answer]
just use a for-loop like normal, final example here for instance https://yalmip.github.io/tutorial/basics/

**Comment by sb1368** [comment]
Many thanks for your help. This example guides me to what I want.
